Migration of db fails when upgrading to 2.14.0

**Bug description**
Error when moving from 2.13.1 docker image to 2.14.0 (latest). Failed migration: while backing up db before migration: read /data/portainer.db: permission denied. Also I do see files in backups/common/ like portainer.db.035.20220628192528 but they are 0 bytes.

I am able to rollback to the image 2.13.1 and access the db fine.

**Expected behavior**
New version migration completes successfully.

**Portainer Logs**
level=info msg="Encryption key file `portainer` not present"
level=info msg="Proceeding without encryption key"
level=info msg="Loading PortainerDB: portainer.db"
level=info msg="Opened existing store" version=35
level=info msg="2022/06/28 19:25:28 [INFO] [database, backup] [message: creating db backup]"
level=info msg="2022/06/28 19:25:28 [INFO] [database, backup] [message: Copying db file from /data/portainer.db to /data/backups/common/portainer.db.035.20220628192528]"
level=info msg="2022/06/28 19:25:28 [ERROR] [database, backup] [message: Failed] [error: read /data/portainer.db: permission denied]"
level=fatal msg="Failed migration: while backing up db before migration: read /data/portainer.db: permission denied"

**Steps to reproduce the issue:**

Pulled new image (latest) using docker-compose and started container.

**Technical details:**

- Portainer version:
- Docker version (managed by Portainer):
- Kubernetes version (managed by Portainer):
- Platform (windows/linux):
- Command used to start Portainer (`docker run -p 9443:9443 portainer/portainer`):
- Browser:
- Use Case (delete as appropriate): Using Portainer at Home, Using Portainer in a Commercial setup.
- Have you reviewed our technical documentation and knowledge base? Yes/No

**Additional context**
Add any other context about the problem here.

--- a/api/datastore/backup.go
+++ b/api/datastore/backup.go
@@ -104,6 +104,9 @@ func (store *Store) backupWithOptions(options *BackupOptions) (string, error) {
 
 	options = store.setupOptions(options)
 
+	store.Close()
+	defer store.Open()
+
 	return options.BackupPath, store.copyDBFile(store.databasePath(), options.BackupPath)
 }
 

--- a/api/datastore/backup.go
+++ b/api/datastore/backup.go
@@ -103,6 +103,7 @@ func (store *Store) backupWithOptions(options *BackupOptions) (string, error) {
 	store.createBackupFolders()
 
 	options = store.setupOptions(options)
+	dbPath := store.databasePath()
 
 	if err := store.Close(); err != nil {
 		return options.BackupPath, fmt.Errorf(
@@ -111,15 +112,17 @@ func (store *Store) backupWithOptions(options *BackupOptions) (string, error) {
 		)
 	}
 
-	if err := store.copyDBFile(store.databasePath(), options.BackupPath); err != nil {
-		return options.BackupDir, err
+	if err := store.copyDBFile(dbPath, options.BackupPath); err != nil {
+		return options.BackupPath, err
 	}
 
-	_, err := store.Open()
-	return options.BackupDir, fmt.Errorf(
-		"error opening datastore after creating backup: %v",
-		err,
-	)
+	if _, err := store.Open(); err != nil {
+		return options.BackupPath, fmt.Errorf(
+			"error opening datastore after creating backup: %v",
+			err,
+		)
+	}
+	return options.BackupPath, nil
 }
 
 // RestoreWithOptions previously saved backup for the current Edition  with options

--- a/api/datastore/backup.go
+++ b/api/datastore/backup.go
@@ -104,10 +104,22 @@ func (store *Store) backupWithOptions(options *BackupOptions) (string, error) {
 
 	options = store.setupOptions(options)
 
-	store.Close()
-	defer store.Open()
+	if err := store.Close(); err != nil {
+		return options.BackupPath, fmt.Errorf(
+			"error closing datastore before creating backup: %v",
+			err,
+		)
+	}
 
-	return options.BackupPath, store.copyDBFile(store.databasePath(), options.BackupPath)
+	if err := store.copyDBFile(store.databasePath(), options.BackupPath); err != nil {
+		return options.BackupDir, err
+	}
+
+	_, err := store.Open()
+	return options.BackupDir, fmt.Errorf(
+		"error opening datastore after creating backup: %v",
+		err,
+	)
 }
 
 // RestoreWithOptions previously saved backup for the current Edition  with options

